buildenv/jenkins/docker-slaves/x86/centOS/Dockerfile

```diff
@@ -0,0 +1,215 @@
+  && tar xfvz ant-contrib-1.0b3-bin.tar.gz -C .\
+  && rm ant-contrib-1.0b3-bin.tar.gz \
+  && cd ant-contrib \
+  && mv ant-contrib-1.0b3.jar .. \
+  && cd .. \
+  && rm -rf ant-contrib
+
+RUN cd /usr/src/ \
+  && wget http://developer.download.nvidia.com/compute/cuda/7.5/Prod/local_installers/cuda_7.5.18_linux.run \
+  && chmod +x cuda_7.5.18_linux.run \
+  && /usr/src/cuda_7.5.18_linux.run --silent --toolkit --override
+
+# add user home/jenkins
+RUN useradd -ms /bin/bash ${USER} \
+  && mkdir /home/${USER}/.ssh/
+
+COPY authorized_keys /home/${USER}/.ssh/authorized_keys
+
+RUN chown -R ${USER}:${USER} /home/${USER} \
+  && chmod 644 /home/${USER}/.ssh/authorized_keys
+
+# set up sshd config
+RUN mkdir /var/run/sshd \
+  && sed -i 's/#PermitRootLogin/PermitRootLogin/' /etc/ssh/sshd_config \
+  && sed -i 's/#RSAAuthentication.*/RSAAuthentication yes/' /etc/ssh/sshd_config \
+  && sed -i 's/#PubkeyAuthentication.*/PubkeyAuthentication yes/' /etc/ssh/sshd_config
+
+# SSH login fix. Otherwise user is kicked off after login
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# Download and install boot JDK from AdoptOpenJDKs
+
+RUN ln -s java-1.7.0 /usr/lib/jvm/java-7-openjdk-amd64
+
+RUN mkdir /usr/lib/jvm/adoptojdk-java-80 \
+  && cd /usr/lib/jvm/adoptojdk-java-80 \
+  && wget -O bootjdk8.tar.gz "https://api.adoptopenjdk.net/v2/binary/releases/openjdk8?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk" \
+  && tar -xzf bootjdk8.tar.gz \
+  && rm -f bootjdk8.tar.gz \
+  && mv $(ls | grep -i jdk) bootjdk8 \
+  && mv bootjdk8/* /usr/lib/jvm/adoptojdk-java-80
+
+RUN mkdir /usr/lib/jvm/adoptojdk-java-90 \
+  && cd /usr/lib/jvm/adoptojdk-java-90 \
+  && wget -O bootjdk9.tar.gz "https://api.adoptopenjdk.net/v2/binary/releases/openjdk9?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk" \
+  && tar -xzf bootjdk9.tar.gz \
+  && rm -f bootjdk9.tar.gz \
+  && mv $(ls | grep -i jdk) bootjdk9 \
+  && mv bootjdk9/* /usr/lib/jvm/adoptojdk-java-90
+  
+RUN mkdir /usr/lib/jvm/adoptojdk-java-10 \
+  && cd /usr/lib/jvm/adoptojdk-java-10 \
+  && wget -O bootjdk10.tar.gz "https://api.adoptopenjdk.net/v2/binary/releases/openjdk10?openjdk_impl=openj9&os=linux&arch=x64&release=latest&type=jdk" \
+  && tar -xzf bootjdk10.tar.gz \
+  && rm -f bootjdk10.tar.gz \
+  && mv $(ls | grep -i jdk) bootjdk10 \
+  && mv bootjdk10/* /usr/lib/jvm/adoptojdk-java-10
+
+RUN cd /home/${USER} \
+  && wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tgz \
+  && tar -xzf freemarker.tgz freemarker-2.3.8/lib/freemarker.jar --strip=2 \
+  && rm -f freemarker.tgz
+
+RUN cd /usr/src \
+  && wget http://curl.haxx.se/download/curl-7.24.0.tar.bz2 \
+  && tar xfj curl-7.24.0.tar.bz2 \
+  && rm -f curl-7.24.0.tar.bz2 \
+  && cd curl-7.24.0 \
+  && ./configure --prefix=/usr \
+  && make \
+  && make install 
+
+RUN mkdir /home/${USER}/openjdk_cache \
```

Comment:
Add comment: `# Set up Git remotes to get JDK source from the extensions repositories.`

Reply:
`# Setup a reference repository cache for faster clones in the container`